docs/src/main/sphinx/connector/delta-lake.rst

```diff
@@ -136,10 +136,10 @@ connector.
       - Description
       - Default
     * - ``delta.domain-compaction-threshold``
-      - Sets the number of transactions to act as threshold. Once reached the
-        connector initiates compaction of the underlying files and the delta
-        files. A higher compaction threshold means reading less data from the
-        underlying data source, but a higher memory and network consumption.
+      - Sets the number of transactions to act as a threshold. After reaching
```

Comment:
"as a threshold during query planning."

Reply:
This description seems wrong - IIRC the compaction threshold basically says that if you have a large `IN` list (doesn't apply to `OR`) then it can be simplified into smaller predicates (the implementation detail doesn't matter - may vary across connectors) but only if the number of values in the `IN` list is < domain-compaction-threshold.

e.g. `IN (1, 2, 3, 4)` with `domain_compaction_threshold=2` would not get optimised. But with `domain-compaction-threshold=10` it would get optimised.

Reply:
Also for user facing docs I as a user would be happy with:

> Some databases perform poorly when a large `IN` list or long `WHERE` clauses are pushed down as a predicate to the data source. To optimise such situations Trino can compact the large predicates. To ensure a balance of performance and pushdown the maximum size of predicates that will be compacted can be controlled using `domain-compaction-threshold`.

Reply:
>  if you have a large predicate (WHERE or IN, doesn't apply to OR) 

we must not be specific about that because e.g. sometimes OR can be replaced with IN.

> e.g. `IN (1, 2, 3, 4)` with `domain_compaction_threshold=2` would not get optimised. But with `domain-compaction-threshold=10` it would get optimised.

The opposite. `domain_compaction_threshold=2` => predicate simplifies to equivalent of `BETWEEN 1 and 4`
with `domain_compaction_threshold=10` it's kept as is.

Reply:
Thanks updated the "Also for user facing docs I as a user would be happy with:" to match.

Reply:
Minor suggested rewording of your proposed wording, for style. Should start with description then move onto justification:

> Maximum number of query predicates that Trino compacts for the data source. Some databases perform poorly when a > large list of predicates is pushed down to the data source, so for optimization Trino can compact large predicates up to > this threshold.

Reply:
This is what I came up with based on everyone's suggestions:

`Minimum size of query predicates above which Trino starts compaction. Some databases perform poorly when a large list of predicates is pushed down to the data source. 
For optimization in that situation, Trino can compact the large predicates.
When necessary, adjust the threshold to ensure a balance between performance and pushdown.
`

Please let me know if this is correct.

Reply:
The confusing thing, to me at least, is that it's not the minimum size for compaction but a maximum number of predicates that are compacted. Per @hashhar 's example, `IN (1, 2, 3, 4)` is not compacted if this property is set to 2, but it is if the property is set to 10. If that's the case, then the word "threshold" is unfortunately very misleading and we need to word around it somehow.

Reply:
It's confusing to me too based on the examples and sizes discussed.  The default value is 100.  @findepi said it was the opposite behavior actually than what @hashhar 's statement. Hashar's description as a minimum threshold makes some sense to me, and threshold is part of the config property's name. I'll wait for another round of reviews. Thank you all for your guidance.